illumos-utils/src/zfs.rs

```diff
@@ -220,6 +220,76 @@ pub struct DestroySnapshotError {
     err: crate::ExecutionError,
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum EnsureDatasetVolumeErrorInner {
+    #[error("{0}")]
```

Comment:
Tiny nit - if there's no useful context for the wrapper, you should use

```suggestion
    #[error(transparent)]
```

If you combine `"{0}"` with `#[from]`, you'll get one level of doublespeak when printing the error chain (because this error type and its first source will both display the `ExecutionError`); `#[error(transparent)]` doesn't have that problem.